Rewrite mount.fuse.ceph (to python) and move ceph-fuse options to fs_mntops

ledgr opened the pull request:
Rewrote mount.fuse.ceph to move ceph-fuse options to `fs_mntops`, where it should be.
Bash version with options in `fs_spec` is counterintuitive and cause issues in some
situations (systemd, for example)

jcsp commented:
I don't think any of our automated testing covers this.  How have you tested it?

ledgr commented:
We have used this script for half a year for our project. Then switched to kernel client.
It works with standard fuse and systemd .mount and .automount units. We didn't do any sophisticated tests.

ledgr commented:
Systemd `unit` file used:

```
[Unit]
Description=Mount /home as ceph-fuse client
DefaultDependencies=yes
After=remote-fs-pre.target network.target network-online.target

[Install]
WantedBy=multi-user.target

[Mount]
What=none
Type=fuse.ceph
Options=ceph.name=client.cephfuse-client.domain.tld,nonempty,_netdev
Where=/home
```

_netdev is necessary: https://www.freedesktop.org/software/systemd/man/systemd.mount.html#Automatic%20Dependencies

>  In some cases this is not sufficient (for example network block device based mounts, such as iSCSI), in which case _netdev may be added to the mount option string of the unit, which forces systemd to consider the mount unit a network mount.

Systemd `automount` unit used:
```
[Unit]
Description=/home ceph-fuse Automount Point
DefaultDependencies=yes
After=remote-fs-pre.target network.target network-online.target

[Automount]
Where=/home
```

jcsp commented:
OK, I think I'm on board with this, but it seems like it wouldn't be hard to make it backward compatible for existing fstabs (i.e. detect if the first column is not none, and parse out the old-style args in this case).

An update to doc/cephfs/fstab.rst is also needed.

jcsp commented:
@ledgr could you add the backward compatibility change and docs update so that we can merge it?  I appreciate this works already in your environment but if we merged it as-is it would break some people's systems and confuse anyone reading the docs.

jcsp commented:
(note to self, don't merge until we branch Kraken)

jcsp commented:
@ledgr are you still interested in coming back to this?

ledgr commented:
@jspc, yes, I will come back to it later this week.

jspc commented:
@ledgr - I think you want @jcsp 😁

ledgr commented:
Haha, yes, a typo 😁

ledgr commented:
@jcsp

ledgr commented:
Updated for backward compatibility with existing fstabs.
Updated docs.

Tested python versions:
```
Python 2.6.6
Python 2.7.9
Python 3.4.5
```

Tested mount versions:
```
mount from util-linux 2.25.2 (libmount 2.25.0: selinux, assert, debug)
mount from util-linux 2.20.1 (with libblkid and selinux support)
mount from util-linux 2.28.2 (libmount 2.28.2: selinux, btrfs, force-mountinfo, assert, debug)
mount from util-linux 2.23.2 (libmount 2.23.0: selinux, debug, assert)
```
CentOS 7.2 had problems with mount: https://bugzilla.redhat.com/show_bug.cgi?id=1248003
Which is fixed by updating `libmount to 2.23.2-33.el7`

@jcsp 
> I don't think any of our automated testing covers this.

I didn't write any tests for this, nor did I test it against multiple `ceph-fuse` versions

jcsp commented:
:+1: thank you @ledgr
